components/brave_wallet/browser/eth_tx_controller.cc

```diff
@@ -153,7 +153,83 @@ void EthTxController::AddUnapprovedTransaction(
     return;
   }
 
-  EthTxStateManager::TxMeta meta(std::make_unique<EthTransaction>(*tx));
+  auto tx_ptr = std::make_unique<EthTransaction>(*tx);
+
+  // Use default gas price and limit for ETHSend.
+  if (tx_ptr->data().empty()) {  // ETHSend
```

Comment:
Might be nicer to call `GetTransactionInfoFromData` from `components/brave_wallet/browser/eth_data_parser.h` and check if the tx_type is ETHSend from this mojo type:
```
enum TransactionType {
  ETHSend,
  ERC20Transfer,
  ERC20Approve,
  Other
};
```
You could allow `GetTransactionInfoFromData` to take in nullptr for the parameter info when not needed and don't parse it in that case.

Reply:
Fixed in https://github.com/brave/brave-core/pull/10131/commits/642734ae3d2a7c4e088c8274d06d34db7bbd2986